nntrainer/src/optimizer.cpp

```diff
@@ -189,4 +189,22 @@ int Optimizer::setProperty(std::vector<std::string> values) {
 
   return status;
 }
+
+void Optimizer::read(std::ifstream &file) {
+  if (type == OptType::adam) {
```

Comment:
If optimizer parameters can be different depending on `OptType`, how about saving/reading opt type as well, at the first byte?

Also, in case of old models without opt parameters, I think you can handle them like that `continue_train` is unsupported, by checking file positions.

Reply:
I dont think I understand completely about support existing models?
But I agree with you to save the `OptType`. This will also help in verifying if the current model being loaded matches with the optimizer type.

Reply:
Ah, I also don't know you need to support old models too. It's just my guess about CI's failures.

Reply:
The CI's failures was due to existing bugs exposed because of this PR.
They have now been resolved with the new push.

Reply:
Great!